src/tactic/core.lean

```diff
@@ -737,6 +737,24 @@ do
     lemmas.any_of (λ H, opt.apply H >> tac)) <|>
   fail "apply_any tactic failed; no lemma could be applied"
 
+/--
+This is a version of `apply_any` that takes a list of `tactic expr`s instead of `expr`s,
+and evaluates these as thunks before trying to apply them.
+
+We need to do this to avoid metavariables getting stuck during subsequent rounds of `apply`.
+-/
+meta def apply_any_thunk
```

Comment:
This duplicates code from `apply_any`. You could implement
```lean
meta def apply_any
  (lemmas : list expr)
  (opt : apply_any_opt := {})
  (tac : tactic unit := skip) : tactic unit :=
apply_any_thunk (lemmas.map pure) opt tac
```

Reply:
Fixed, sorry.